src/material.cpp

```diff
@@ -469,13 +499,176 @@ void pyne::Material::write_hdf5(std::string filename, std::string datapath,
+  //        - NO -> CONTINUE
+  //          - "/material" EXIST:
+  //            - NO -> create "/material" group and write everyting in it
+  //            - YES:
+  //              - "/material" is a DATASET -> detect NUCPATH + old write_hdf5
+  //              - "/material" is a GROUP -> add data in it
+  //              - "/material" isn't a GROUP nor a DATASET -> fail and complain
+
+  hid_t db;
+  if (!pyne::file_exists(filename)) {
+    // Create the file
+    db = H5Fcreate(filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
+
+  } else {
+    bool ish5 = H5Fis_hdf5(filename.c_str());
+    if (!ish5) throw h5wrap::FileNotHDF5(filename);
+    db = H5Fopen(filename.c_str(), H5F_ACC_RDWR, fapl);
+
+    // there are a few things that may be true here:
+    // datapath does not exist - there are many possible reasons for this, including
+    //         datapath = "non_default_path" without a leading slash (/) --> assume this is a modifier for a new format
+    // datapath does exist:
+    //         datapath = "/non_default_path" --> old format
+    //         datapath = "/material"   --> new format
+    if (h5wrap::path_exists(db, datapath) && datapath != "/material") {
+      std::string nucpath;
+      hid_t data_set = H5Dopen2(db, datapath.c_str(), H5P_DEFAULT);
+
+      // Grab the nucpath
+      if (detect_nuclidelist(data_set, nucpath)) {
+      deprecated_write_hdf5(db, datapath, "/nucid", row, chunksize);
+      H5Fclose(db);
+        return;
+      } else {  // can't find a valid nuclide list path form datapath... fail
+        throw std::runtime_error(
+            "Can't find the nuclide list path in the existing datapath. Can't "
+            "add your material to the datapath.");
+      }
+    }
+  }
+  
+  // Check if /material exist and what type it is
+  herr_t status;
+  H5O_info_t object_info;
+  status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+  status = H5Oget_info_by_name(db, "/material", &object_info, H5P_DEFAULT);
+  // check if "/material" exists
+  if (status == 0) {
+    if (object_info.type ==
+        H5O_TYPE_DATASET) {  // "/material" is a dataset -> old format use
+                             // deprecated write_hdf5
+      deprecated_write_hdf5(db, datapath, "/nucid", row, chunksize);
+      H5Fclose(db);
+      return;
+
+    } else if (object_info.type != H5O_TYPE_GROUP) {
+      // "/material" either a dataset or a group: fail!
+      throw std::runtime_error("Non-group/non-dataset object /material already exists in the file. Can't write the Material");
+    } else {  // "/material" is a group get his hid
+      material_grp_id = H5Gopen2(db, "/material", H5P_DEFAULT);
+    }
+  } else {  // "/material" do not exist -> create it !
+    material_grp_id =
+        H5Gcreate2(db, "/material", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+  }
+  // Check if /material exist as a Group
+  // Group "/material/datapath" does not exist create it
+  if (!h5wrap::path_exists(db, "/material" + datapath)) {
```

Comment:
Isn't your default `datapath` still "/material"?  So doesn't this mean we get "/material/material" as the default group?

Reply:
do you think we should use an other default ? like "mat" ?

Reply:
or "mat_name" or something to make sure it's clear, at least to the software developers...